src/coprocessor/codec/batch/lazy_column.rs

```diff
@@ -156,28 +125,16 @@ impl LazyBatchColumn {
     #[inline]
     pub fn capacity(&self) -> usize {
         match self {
-            LazyBatchColumn::Raw(ref v) => v.capacity(),
-            LazyBatchColumn::Decoded(ref v) => v.capacity(),
+            LazyBatchColumn::Raw(v) => v.capacity(),
+            LazyBatchColumn::Decoded(v) => v.capacity(),
         }
     }
 
     #[inline]
-    pub fn retain_by_index<F>(&mut self, mut f: F)
-    where
-        F: FnMut(usize) -> bool,
-    {
+    pub fn retain_by_array(&mut self, retain_arr: &[bool]) {
```

Comment:
Would be better if the method here also has doc comments (although it seems repeated)